perllib/FixMyStreet/Roles/CobrandSLWP.pm

```diff
@@ -204,17 +204,36 @@ sub available_permissions {
 sub clear_cached_lookups_property {
     my ($self, $id) = @_;
 
+    # Need to call this before clearing GUID
+    $self->clear_cached_lookups_bulky_slots($id);
+
     foreach my $key (
         $self->council_url . ":echo:look_up_property:$id",
         $self->council_url . ":echo:bin_services_for_address:$id",
-        $self->council_url . ":echo:available_bulky_slots:earlier:$id",
-        $self->council_url . ":echo:available_bulky_slots:later:$id",
         $self->council_url . ":echo:bulky_event_guid:$id",
     ) {
         delete $self->{c}->session->{$key};
     }
 }
 
+sub clear_cached_lookups_bulky_slots {
+    my ( $self, $id ) = @_;
+
+    for (qw/earlier later/) {
+        delete $self->{c}->session->{ $self->council_url
+                . ":echo:available_bulky_slots:$_:$id" };
+    }
+
+    # We also need to cancel the reserved slots in Echo
+    my $guid_key = $self->council_url . ":echo:bulky_event_guid:$id";
+    my $guid = $self->{c}->session->{$guid_key};
+
+    my $cfg = $self->feature('echo');
+    my $echo = Integrations::Echo->new(%$cfg);
+
+    $echo->CancelReservedSlotsForEvent($guid);
```

Comment:
The first time this is called `$guid` is empty and the resulting SOAP call returns HTTP 500 - we should probably skip the call in that case.

Reply:
(also this ends up getting called just before the payment screen somehow, when it probably doesn't need to be, as the event is about to be created in Echo [which releases](https://docs.google.com/document/d/1oEXsphrObtoXnf_o_EMBbnjmYO9roWL5/edit#bookmark=id.f8joworzs9l) the reserved slots)

Reply:
Is it an issue that it gets called just before payment? Looks like it's due to the `clear_cached_lookups_property` call under `add_report` in perllib/FixMyStreet/App/Controller/Waste.pm.

Reply:
Maybe, maybe not. The call seemed to succeed when I tried it at least.

I suppose there is a small chance that the following occurs:

 1. user has a reservation on the final available slot for a date
 2. which is then released by the `CancelReservedSlotsForEvent` call just before they pay
 3. another user gets a reservation on that last slot which was just released
 4. The `PostEvent` for the first user's booking happens (by the report being sent to open311-adapter and submitted to Echo) but the slot was already taken
 5. I don't know what happens next?

In fairness there should only be a few seconds between the events of steps 2 and 4 so the chance of step 3 happening is very very small. Either way, if there's a simple way to skip the call if it happens from `add_report` it is probably worth doing.